ecosystem/sep-0024.md

```diff
@@ -190,6 +191,72 @@ The deposit flow can only be fulfilled if the Stellar `Account` has established
+
+**Wallets** poll for their users' received Claimable Balances so that users can "claim" them.
+
+**Anchors** can optionally use this functionality to send SEP-24 deposit transactions to Stellar accounts that don't yet have a trust-line to the asset they have requested.
+
+**Wallet Claimable Balance Flow**
+
+Wallets will still follow the [typical deposit flow](#Basic-Wallet-Implementation).
+The POST request however will extend to include a `claimable_balance` parameter
+
+```
+POST TRANSFER_SERVER_SEP0024/transactions/deposit/interactive
+Content-Type: multipart/form-data
+```
+
+New Request Parameter:
+
+Name | Type | Description
+-----|------|------------
+`claimable_balance` | boolean | True if submitting transactions as a claimable balance is accepted, false otherwise.
+
+**Wallet Claimable Balance Polling**
+
+The wallet must either listen for a callback from the anchor with the
+`claimable_balance_id` or poll Horizon's
+[`/claimable_balances/`](https://developers.stellar.org/api/resources/claimablebalances/list/)
+endpoint to pull any available claimable balances for the claimant.
+
+*Recommended request*
+`GET /claimable_balances?claimant={:claimant}&order={asc,desc}&limit={1-200}`
+
+This allows the wallet to update the claimant that their is a balance available
+to claim and they may need to establish a [trust-line](#stellar-account-doesnt-trust-asset) in order to do so.
+
+**Anchor Claimable Balance Flow**
+
+1. The given Stellar `account` does not have a trust-line to the asset's issuer account.
+2. The `Anchor` proceeds with the deposit flow via the [CAP-0023](../core/cap-0023.md) Create Claimable Balance operation.
+
+**Anchor Claimable Balance Field Changes**
+
+### Fields for deposit transactions
+
+Name | Type | Description
+-----|------|------------
+`claimable_balance_id` | string | (optional) ClaimableBalanceID is a required parameter when claiming a Claimable Balance operation.
+
+The `claimable_balance_id` can be added to the [response](#deposit-and-withdraw-shared-responses)
+This can be helpful for wallets to communicate to the user that their
+transaction was submitted as a claimable balance and can be claimed at the relevant location (`claimable_balance_id`).
+
+In order to claim it the recipient must either get access to the account that the anchor created and or establish a trust-line (for the asset code of the claimableBalance) with their existing account.
+
+**Anchor Claimable Balance Recommended Options**
+
+#### Claimable Balance Predicates
+
+[https://developers.stellar.org/docs/glossary/claimable-balance/#parameters](https://developers.stellar.org/docs/glossary/claimable-balance/#parameters)
+
+We suggest if you decide to set predicates to claim the balance in 30 days. `BEFORE_RELATIVE_TIME(X)`
+
+However if implemented, its up to the anchors to handle refunds (of off network funds) to the the waller/user.
```

Comment:
I think we need to expand on this. _why_ does this introduce the need to handle refunds? Why would an anchor want to have an expiration? Why do we recommend an expiration? This SEP is the introduction to this functionality for most people reading this.

Reply:
Correct me if I'm off the mark on this theoretical flow if the anchor sets the 30 day predicate.

1. Alice deposits off chain funds (US dollar)
2. Alice doesn't have a trust-line established for (USD) such that the anchor makes a claimable balance for Alice to claim (30 days to claim)
3. Alice doesnt claim the balance before 30 days and the balance is locked.
4. The anchor should refund Alice of her off chain funds

Reply:
Honestly after writing that and demoing what that behavior would be I think its probably for the best if we ensure that they set the predicates as unconditional.

Reply:
I don't think we should recommend going either way, but outline why an anchor would take one approach over the other. 

Anchors that really really care about UX will probably want to handle refunds so users don't think they stole funds only to realize it's on them to "claim" the balance. 

For other anchors, handling refunds won't be worth it because there is no real benefit to them if they refund. The have to do 2X the logistical effort for no profit (fees are refunded).

Reply:
@tomerweller 
Would you be opposed to strongly suggesting unconditional claiming conditions?